Sources/Classes/RSContext.m

```diff
@@ -174,16 +174,27 @@ - (void)putAdvertisementId:(NSString *_Nonnull)idfa {
     dispatch_sync(queue, ^{
         if( idfa != nil && [idfa length] != 0) {
             [RSLogger logDebug:[[NSString alloc] initWithFormat:@"IDFA: %@", idfa]];
+            [[RSPreferenceManager getInstance] saveAdvertisingId:idfa];
```

Comment:
There is an edge case that we need to handle: 

- AdvertisingId should only be saved if it is a valid value i.e., if the `idfa` value is equal to this `00000000-0000-0000-0000-000000000000` then do not save it.

Otherwise, we will get unexpected behaviour. Follow this:
```
[RSClient putAdvertisingId:@"00000000-0000-0000-0000-000000000000"];
// SDK init
// Other events
```
- At the first app launch everything will be okay with the events payload.
```
        "device": {
            "attTrackingStatus": 0,
            "manufacturer": "Apple",
            "model": "arm64",
            "name": "iPhone 15 Pro",
            "type": "iOS"
        },
```
- But, at the next launch, you can see the difference in the payload:
```
      "device": {
            "adTrackingEnabled": true,
            "advertisingId": "00000000-0000-0000-0000-000000000000",
            "attTrackingStatus": 0,
            "manufacturer": "Apple",
            "model": "arm64",
            "name": "iPhone 15 Pro",
            "type": "iOS"
        },
```

Please check if the same issue could happen in Android or not.

Reply:
Agreed, pushed a fix to save advertisingId to defaults only if its valid.